src/types/input_method_v2.zig

```diff
@@ -0,0 +1,107 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+pub const InputMethodManagerV2 = extern struct {
+    global: *wl.Global,
+    input_methods: wl.list.Head(InputMethodV2, "link"),
+
+    server_destroy: wl.Listener(*wl.Server),
+
+    events: extern struct {
+        input_method: wl.Signal(*wlr.InputMethodV2),
+        destroy: wl.Signal(*wlr.InputMethodManagerV2),
+    },
+
+    extern fn wlr_input_method_manager_v2_create(server: *wl.Server) ?*wlr.InputMethodManagerV2;
+    pub fn create(server: *wl.Server) !*wlr.InputMethodManagerV2 {
+        return wlr_input_method_manager_v2_create(server) orelse error.OutOfMemory;
+    }
+};
+
+pub const ImputMethodV2 = extern struct {
+    pub const PreeditString = extern struct {
+        text: [*:0]u8,
+        cursor_begin: i32,
+        cursor_end: i32,
+    };
+
+    pub const DeleteSurroundingText = extern struct {
+        before_length: u32,
+        after_length: u32,
+    };
+
+    pub const State = extern struct {
+        preedit: wlr.InputMethodV2.PreeditString,
+        commit_text: [*:0]u8,
+        delete: wlr.ImputMethodV2.DeleteSurroundingText,
+    };
+
+    pub const KeyboardGrab = extern struct {
+        ressource: *wl.Ressource,
+        input_method: *wlr.InputMethodV2,
+        keyboard: *wlr.Keyboard,
+
+        keyboard_keymap: wl.Listener(*wlr.Keyboard),
+        keyboard_repeat_info: wl.Listener(*wlr.Keyboard),
+        keyboard_destroy: wl.Listener(*wlr.Keyboard),
+
+        events: extern struct {
+            destroy: wl.Signal(*wlr.InputMethodV2.KeyboardGrab),
+        },
+
+        extern fn wlr_input_method_keyboard_grab_v2_send_key(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab, time: u32, key: u32, state: u32) void;
+        pub const sendKey = wlr_input_method_keyboard_grab_v2_send_key;
+
+        extern fn wlr_input_method_keyboard_grab_v2_send_modifiers(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab, modifiers: *wlr.Keyboard.Modifiers) void;
+        pub const sendModifiers = wlr_input_method_keyboard_grab_v2_send_modifiers;
+
+        extern fn wlr_input_method_keyboard_grab_v2_set_keyboard(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab, keyboard: *wlr.Keyboard) void;
+        pub const setKeyboard = wlr_input_method_keyboard_grab_v2_set_keyboard;
+
+        extern fn wlr_input_method_keyboard_grab_v2_destroy(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab) void;
+        pub const destroy = wlr_input_method_keyboard_grab_v2_destroy;
+    };
+
+    ressource: *wl.Ressource,
```

Comment:
```suggestion
    resource: *wl.Resource,
```

Reply:
Damn I'm sorry I write it in french everywhere